api/v1/controllers/utils.js

```diff
@@ -46,6 +46,19 @@ function hasDuplicates(tagsArr) {
 }
 
 /**
+ * Check if name field is in the object.
+ * Throws validation error if object contains name field.
+ * @param  {Object} obj - Request object
+ */
+function rejectIfNameInBody(obj) {
```

Comment:
I think this function is only useful in the sample model. For every other model, you would just use the "noReadOnlyFieldsInReq" function. This one, presumably, is special because the name field is read-only for most things but it's NOT read-only for **upserts**.
So... maybe pull this fn back into the samples controller instead of utils, since it should never be used by any other model?